storage/aptosdb/src/pruner/state_store/mod.rs

```diff
@@ -36,33 +29,25 @@ impl DBPruner for StateStorePruner {
         STATE_STORE_PRUNER_NAME
     }
 
-    fn prune(&self, _ledger_db_batch: &mut SchemaBatch, max_versions: u64) -> Result<Version> {
+    fn prune(&self, _ledger_db_batch: &mut SchemaBatch, batch_size: usize) -> Result<Version> {
         if !self.is_pruning_pending() {
             return Ok(self.min_readable_version());
         }
         let min_readable_version = self.min_readable_version.load(Ordering::Relaxed);
         let target_version = self.target_version();
-        return match prune_state_store(
-            &self.db,
+        return match self.prune_state_store(
             min_readable_version,
```

Comment:
Do you still need to take min_readable_version and target_version as input parameter, given you've already moved the function into the struct?

Reply:
It is ugly but we need them for testing.